src/lib/chart/dendrogram.typ

```diff
@@ -0,0 +1,158 @@
+#import "../axes.typ"
+#import "../palette.typ"
+#import "../../draw.typ"
+
+#let dendrogram-default-style = (
+  axes: (tick: (length: 0))
+)
+
+// Valid dendrogram modes
+#let dendrogram-modes = (
+  "basic",
+)
+
+// Functions for max value calculation
+#let dendrogram-max-value-fn = (
+  basic: (data, value-key) => {
+    calc.max(0, ..data.map(t => t.at(value-key)))
+  },
+)
+
+// Functions for min value calculation
+#let dendrogram-min-value-fn = (
+  basic: (data, value-key) => {
+    calc.min(0, ..data.map(t => t.at(value-key)))
+  },
+)
+
+// TODO:
+
+#let dendrogram(data,
+                 x1-key: 0,
+                 x2-key: 1,
+                 height-key: 2,
+                 mode: "basic",
```

Comment:
If there is only one mode, remove the mode setting.

Reply:
I left that in there for the eventual implementation of the horizontal dendrogram, so that the function signatures don't change when its implemented

Reply:
Ok, that's fine.